### PR TITLE
Add mocks

### PR DESCRIPTION
**Pull request by gouthamMN**
**What does this PR do?**
- Generate test mocks that can be used for testing.
- Add target `generate` to Makefile.
- Update bingo to install mockgen tool.
- Update unit test to run parallelly.

### CODE REVIEW

File: client/generate.go [hunk 1]
```diff
@@ -3,6 +3,6 @@ package client
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
-//go:generate rm -rf ../../../../../mocks/azureclient/authz/$GOPACKAGE
-//go:generate mockgen -destination=../../../mocks/azureclient/authz/$GOPACKAGE/$GOPACKAGE.go github.com/Azure/ARO-RP/pkg/util/azureclient/authz/$GOPACKAGE RemotePDPClient
-//go:generate goimports -local=github.com/Azure/ARO-RP -e -w ../../../mocks/azureclient/authz/$GOPACKAGE/$GOPACKAGE.go
+//go:generate rm -rf mocks/$GOPACKAGE
+//go:generate mockgen -destination=mocks/$GOPACKAGE/$GOPACKAGE.go github.com/Azure/checkaccess-v2-go-sdk/client RemotePDPClient
```
Comment (stevekuznetsov) [inline client/generate.go:7]: How are you sure this is using the correct `$(MOCKGEN)` from the `.bingo` install?

File: client/generate.go [hunk 2]
```diff
@@ -3,6 +3,6 @@ package client
...
+//go:generate goimports -local=github.com/Azure/checkaccess-v2-go-sdk -e -w mocks/$GOPACKAGE/$GOPACKAGE.go
```
Comment (stevekuznetsov) [inline client/generate.go:8]: I think you also need to version `goimports`, no?

File: client/generate.go (hunk 1 above)
Comment (gouthamMN) [reply client/generate.go:7]: Yes.

File: client/generate.go (hunk 2 above)
Comment (gouthamMN) [reply client/generate.go:8]: updated

File: client/generate.go (hunk 1 above)
Comment (stevekuznetsov) [reply client/generate.go:7]: My question was - _how_ are you sure? The `mockgen` binary created by `bingo` is not literally called `mockgen` - it's `$(GOBIN)/mockgen-v0.5.0`. Try running `command -v mockgen` to see if there is such a binary in your path that `go generate` is using instead.

**Pull request closed without merging**